src/io.jl

```diff
@@ -22,64 +21,115 @@ function load(f::File{DataFormat{:PNG}})
 
     width = png_get_image_width(png_ptr, info_ptr)
     height = png_get_image_height(png_ptr, info_ptr)
-    color_type = png_get_color_type(png_ptr, info_ptr)
-    bit_depth = png_get_bit_depth(png_ptr, info_ptr)
+    color_type_orig = png_get_color_type(png_ptr, info_ptr)
+    color_type = color_type_orig
+    bit_depth_orig = png_get_bit_depth(png_ptr, info_ptr)
+    bit_depth = bit_depth_orig
     num_channels = png_get_channels(png_ptr, info_ptr)
     interlace_type = png_get_interlace_type(png_ptr, info_ptr)
-    buffer_eltype = _buffer_color_type(color_type, bit_depth)
 
-    @debug(
-        "Read PNG info:",
-        f.filename,
-        height,
-        width,
-        color_type,
-        bit_depth,
-        num_channels,
-        interlace_type,
-        buffer_eltype
-    )
+    backgroundp = png_color_16p()
+    if png_get_bKGD(png_ptr, info_ptr, Ptr{png_color_16p}(backgroundp)) != 0
+        png_set_background(png_ptr, backgroundp, PNG_BACKGROUND_GAMMA_FILE, 1, 1.0)
+    end
```

Comment:
Pretty sure this does not work properly (when I comment this out for images with background chunks, this doesn't change the output). It would be cool to get to a stage where we can see that the underlying backgorund (`backgroundp` is a pointer) got actually updated after the `png_get_bKGD` call. I was experimenting with mutable structs and redefining `png_get_bKGD` and `png_set_background` a bit, but with little success.